### PR TITLE
[CPU] Fix sdpa pattern match when adding result

### PR DESCRIPTION
**Pull request by luo-cheng2021**
### Details:
 - *Adding results to a stateful model leads to a crash.*
 - Add test case for(Depending on #24954):
 ```
ReadValue->Gather->ShapeOf->...
               |------->Concat->SDPA...
```

### Tickets:
 - *[143098](https://jira.devtools.intel.com/browse/CVS-143098)*

### CODE REVIEW

**Comment by dmitry-gorokhov** [comment]
@maxnick Please review the PR

File: src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/stateful_sdpa_fusion.cpp
Comment (maxnick) [inline src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/stateful_sdpa_fusion.cpp:1]: Shouldn't we extend the existing transformation tests to cover the changes?

File: src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/stateful_sdpa_fusion.cpp [hunk 1]
```diff
@@ -53,9 +53,12 @@ StatefulSDPAFusion::StatefulSDPAFusion() {
     auto concat_k = makePattern<opset1::Concat>({gather_input_k, cur_k}, {{"axis", axis_seq_len}});
     auto concat_v = makePattern<opset1::Concat>({gather_input_v, cur_v}, {{"axis", axis_seq_len}});
 
-    auto multi_query_bcst = [](std::shared_ptr<Node> kv) {
-        auto reshape_kv = wrap_type<opset6::Reshape>({kv, any_input()});
-        auto unsqueeze_kv = makePattern<opset1::Unsqueeze>({kv, any_input()});
+    std::shared_ptr<Node> reshape_k, reshape_v, unsqueeze_k, unsqueeze_v;
+    std::shared_ptr<Node> computed_bcst_k, computed_bcst_v, multiply_k, multiply_v;
+    auto multi_query_bcst = [](const std::shared_ptr<Node>& kv, std::shared_ptr<Node>& reshape_kv,
+        std::shared_ptr<Node>& unsqueeze_kv, std::shared_ptr<Node>& computed_bcst, std::shared_ptr<Node>& multiply_kv) {
```
Comment (maxnick) [inline src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/stateful_sdpa_fusion.cpp:59]: NIT. Isn't it more intuitive to return a tuple with the original return type and all the newly added pointers? Something like that:
```cpp
    std::shared_ptr<Node> reshape_k, reshape_v, unsqueeze_k, unsqueeze_v;
    std::shared_ptr<Node> computed_bcst_k, computed_bcst_v, multiply_k, multiply_v;
    std::shared_ptr<Node> mq_reshape_k;
    std::tie(mq_reshape_k, reshape_k, unsqueeze_k, computed_bcst_k, multiply_k) = multi_query_bcst(concat_k);
```

File: src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/stateful_sdpa_fusion.cpp [hunk 2]
```diff
@@ -181,12 +186,30 @@ StatefulSDPAFusion::StatefulSDPAFusion() {
...
+                }
+            }
+            return true;
+        };
+        if (!is_optional_one_child({convert_past_k, convert_past_v, transpose_q, transpose_k, transpose_v,
+                                    reshape_k, unsqueeze_k, computed_bcst_k, multiply_k,
+                                    reshape_v, unsqueeze_v, computed_bcst_v, multiply_v,
+                                    mq_reshape_k, mq_reshape_v}))
+            return false;
+
```
Comment (maxnick) [inline src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/stateful_sdpa_fusion.cpp:204]:
```suggestion
        if (!is_optional_one_child({convert_past_k, convert_past_v, transpose_q, transpose_k, transpose_v,
                                    reshape_k, unsqueeze_k, computed_bcst_k, multiply_k,
                                    reshape_v, unsqueeze_v, computed_bcst_v, multiply_v,
                                    mq_reshape_k, mq_reshape_v})) {
            return false;
        }

```

**Review by maxnick — CHANGES_REQUESTED** [review]

File: src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/stateful_sdpa_fusion.cpp
Comment (luo-cheng2021) [reply src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/stateful_sdpa_fusion.cpp:1]: Added.

File: src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/stateful_sdpa_fusion.cpp (hunk 1 above)
Comment (luo-cheng2021) [reply src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/stateful_sdpa_fusion.cpp]: Updated.

File: src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/stateful_sdpa_fusion.cpp (hunk 2 above)
Comment (luo-cheng2021) [reply src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/stateful_sdpa_fusion.cpp]: Changed.

**Review by maxnick — APPROVED** [review]

File: src/plugins/intel_cpu/tests/unit/transformations/state_concat_sdpa.cpp [hunk 3]
```diff
@@ -158,3 +194,26 @@ TEST(TransformationTests, StateConcatSDPAMixtral) {
         ASSERT_TRUE(res.first) << res.second;
     }
 }
+
+TEST(TransformationTests, StateConcatSDPAWithExtraNode) {
+    // when some unexpected extra nodes exist in SDPA, the fusion should be fail
```
[the remark this answers is not in the document]
Comment (luo-cheng2021) [reply src/plugins/intel_cpu/tests/unit/transformations/state_concat_sdpa.cpp]: Updated.

**Pull request merged**